### PR TITLE
Create a Github repository integration

### PR DESCRIPTION
**Pull request by gzussa**
### What does this PR do?

A brief description of the change being made with this pull request.

### Motivation

What inspired you to submit this pull request?

### Review checklist

- [ ] PR has a [meaningful title](https://github.com/DataDog/integrations-core/blob/master/CONTRIBUTING.md#pull-request-title) or PR has the `no-changelog` label attached
- [ ] Feature or bugfix has tests
- [ ] Git history is clean
- [ ] If PR impacts documentation, docs team has been notified or an issue has been opened on the [documentation repo](https://github.com/DataDog/documentation/issues/new)

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

**Comment by gzussa** [comment]
note: the token in these changes doesn't exist anymore

**Comment by ruthnaebeck** [comment]
Assuming this is a WIP and adding that label

File: github_repo/README.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# Agent Check: Github_repo
+
+## Overview
+
+This check monitors [Github_repo][1] through the Datadog Agent.
+
+## Setup
+
+### Installation
```
[the remark this answers is not in the document]
Comment (gzussa) [reply github_repo/README.md]: I will regenerate it then

File: github_repo/datadog_checks/github_repo/data/conf.yaml.example [hunk 2]
```diff
@@ -0,0 +1,22 @@
+init_config:
+
+    ## @param access_token - string - required
+    ## The GitHub access token to authenticate to Github APIs
+    #
+  - access_token: <ACCESS_TOKEN>
```
[the remark this answers is not in the document]
Comment (gzussa) [reply github_repo/datadog_checks/github_repo/data/conf.yaml.example]: That's just specific to the workshop.

File: github_repo/datadog_checks/github_repo/github_repo.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        # NOTE: custom_tags is a stretch
+        tags = instance.get('custom_tags', [])
+
+        # Get repository
+        g = Github(self.access_token)
+
+        try:
+            repo = g.get_repo(repository_name)
+            self.log.debug(repo.full_name)
+            tags.append(repository_name)
```
[the remark this answers is not in the document]
Comment (gzussa) [reply github_repo/datadog_checks/github_repo/github_repo.py]: Cleaner indeed 👍

[the remark this answers is not in the document]
Comment (gzussa) [reply github_repo/datadog_checks/github_repo/github_repo.py]: Good catch!

File: github_repo/datadog_checks/github_repo/github_repo.py [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+            watchers = repo.get_watchers().totalCount
+            self.gauge('github_repo.watchers', watchers, tags=tags)
+            contributors = repo.get_contributors().totalCount
+            self.gauge('github_repo.contributors', contributors, tags=tags)
+            subscribers = repo.get_subscribers().totalCount
+            self.gauge('github_repo.subscribers', subscribers, tags=tags)
+
+            if self.since is None:
+                # We need to warm the cache
+                commits = repo.get_commits()
```
[the remark this answers is not in the document]
Comment (gzussa) [reply github_repo/datadog_checks/github_repo/github_repo.py]: That's why there is a caching mechanism. For the purpose of this workshop I won't implement a limit.

File: github_repo/manifest.json [hunk 5]
```diff
@@ -0,0 +1,30 @@
+{
+  "display_name": "Github Repo",
+  "maintainer": "gregory.zussa@datadoghq.com",
+  "manifest_version": "1.0.0",
+  "name": "github_repo",
+  "metric_prefix": "github_repo.",
+  "metric_to_check": "github_repo.up",
```
[the remark this answers is not in the document]
Comment (gzussa) [reply github_repo/manifest.json]: I will pick `contributors` instead.

File: github_repo/metadata.csv [hunk 6]
```diff
@@ -0,0 +1,6 @@
+metric_name,metric_type,interval,unit_name,per_unit_name,description,orientation,integration,short_name
+github_repo.commits,gauge,,unit,,The number of repository commits,0,github_repo,commits
+github_repo.contributors,gauge,,unit,,The number of repository contributors,0,github_repo,contributors
```
[the remark this answers is not in the document]
Comment (gzussa) [reply github_repo/metadata.csv]: Good catch thanks

File: github_repo/tox.ini [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+basepython = py37
+envlist =
+    py{27,37}-github_repo
+
+[testenv]
+dd_check_style = true
+usedevelop = true
+platform = linux|darwin|win32
+deps =
+    datadog-checks-base[deps]>=6.6.0
```
[the remark this answers is not in the document]
Comment (gzussa) [reply github_repo/tox.ini]: I will regenerate the whole project seems i didn't use the latest ddev cli

**Pull request closed without merging**